### PR TITLE
Add integration tests of CustomerSync.

### PR DESCRIPTION
**Pull request by ahmetoz**
Provided main customer sync class integration tests.

- Sync integration tests and check code coverage is checked (source to CTP target and CTP source to CTP target)
- Test scenarios are not minimalistic and have all kinds of reference actions including customer groups, custom fields, and stores.

Related issue: #579

### CODE REVIEW

File: src/integration-test/java/com/commercetools/sync/integration/commons/utils/CustomerITUtils.java [hunk 1]
```diff
@@ -19,6 +57,118 @@ public static void deleteCustomers(@Nonnull final SphereClient ctpClient) {
...
+        @Nonnull final SphereClient ctpClient) {
+
+        final Store storeBerlin = createStore(ctpClient, "store-berlin");
+        final Store storeHamburg = createStore(ctpClient, "store-hamburg");
+        final Store storeMunich = createStore(ctpClient, "store-munich");
+
+        final Type customTypeGoldMember = createCustomerCustomType("customer-type-gold", Locale.ENGLISH,
+            "gold customers", ctpClient);
+
+        final CustomerGroup customerGroupGoldMembers = createCustomerGroup(ctpClient, "gold members", "gold");
```
Comment (lojzatran) [inline src/integration-test/java/com/commercetools/sync/integration/commons/utils/CustomerITUtils.java:70]: This 5 resources could be created in parallel to speed up the whole testing.

**Review by salander85 — APPROVED** [review]

File: src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/customers/CustomerSyncIT.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        assertThat(exceptions).isEmpty();
+
+        assertThat(customerSyncStatistics).hasValues(2, 1, 0, 0);
+        assertThat(customerSyncStatistics
+            .getReportMessage())
+            .isEqualTo("Summary: 2 customers were processed in total (1 created, 0 updated and 0 failed to sync).");
+    }
+
+    @Test
+    void sync_WithUpdates_ShouldReturnProperStatistics() {
```
Comment (lojzatran) [inline src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/customers/CustomerSyncIT.java:104]: Do you want to check if customer was updated correctly or you have this already in a different test?

File: src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/customers/CustomerSyncIT.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            .errorCallback((exception, oldResource, newResource, actions) -> {
+                errorMessages.add(exception.getMessage());
+                exceptions.add(exception);
+            })
+            .build();
+        customerSync = new CustomerSync(customerSyncOptions);
+    }
+
+    @Test
+    void sync_WithoutUpdates_ShouldReturnProperStatistics() {
```
Comment (lojzatran) [inline src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/customers/CustomerSyncIT.java:79]: In the other CustomerSyncIT, you have 2 different cases: `sync_WithNewCustomer_ShouldCreateCustomer` and `sync_WithSameCustomer_ShouldNotUpdateCustomer` -  should you have also these cases here?

**Review by lojzatran — CHANGES_REQUESTED** [review]

File: src/integration-test/java/com/commercetools/sync/integration/commons/utils/CustomerITUtils.java (hunk 1 above)
Comment (ahmetoz) [reply src/integration-test/java/com/commercetools/sync/integration/commons/utils/CustomerITUtils.java:70]: I was aware of this but then forgot about it, good catch :)

File: src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/customers/CustomerSyncIT.java (hunk 3 above)
Comment (ahmetoz) [reply src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/customers/CustomerSyncIT.java:79]: I could add but in the end, the examples in integration tests cover all, less it better to not increase built time.

**Review by lojzatran — APPROVED** [review]

**Pull request merged**